### ISSUE TITLE
ui.Rect doesn't support slicing

### ISSUE THREAD

**Issue by controversial** [omz/Pythonista-Issues#112]
`ui.Rect` supports indexing but not slicing:
```python
>>> a = ui.Slider().frame
>>> a
Rect(0.00, 0.00, 100.00, 34.00)
>>> a[2]
100.0
>>> a[2:]
Traceback (most recent call last):
  File "<string>", line 1, in <module>
TypeError: sequence index must be integer, not 'slice'
```

**Comment by Phuket2** [comment]
I am not sure about my comment here. But you can slice.
It's just should slicing be natural to the ui.Rect object. Controversial  we had discussed this very thing with opetaor overloading. It comes down to what makes sense?

```
f = (0, 0, w, h)
	r = ui.Rect(*f)
	print('tuple', r.as_tuple()[:2])
	print('tuple', r.as_tuple()[-1])
```

**Comment by omz** [comment]
The way slicing works has changed a bit in Python 3 – in Python 2 mode, you can actually slice `ui.Rect` objects directly because it just requires the type to be indexable (i.e. anything that is a "sequence" can be sliced automatically). In Python 3, it's apparently a bit different, but I haven't really looked at the details yet.

**Comment by controversial** [comment]
@Phuket2 Yeah. This counts as a good use of the attributes. It's trying to emulate tuple. 

(Adding subviews to a superview with `+` isn't. 😉 Merging two views might be a good use, though)

**Comment by controversial** [comment]
@omz `Rect` should inherit from `collections.MutableSequence`

**Comment by Phuket2** [comment]
Lol, I am glad I am not a manic depressive sort of guy 😱 I basically get nothing really correct.

**Comment by controversial** [comment]
Nah, it's how learning happens. You get a lot right. I'm learning too, I
get a lot of stuff wrong.

**Comment by dgelessus** [comment]
@controversial `collections.MutableSequence` isn't right either, that would imply that you can use things like `append` and `del` to change its length. This isn't the case, since a `ui.Rect` always has exactly 4 elements.

**Comment by Phuket2** [comment]
It's like the davinci code

**Comment by controversial** [comment]
Oh @dgelessus
